src/tests/unit/tests/common/components/issue-filing-button.test.tsx

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { shallow } from 'enzyme';
+import { DefaultButton } from 'office-ui-fabric-react';
+import * as React from 'react';
+import { IMock, It, Mock, Times } from 'typemoq';
+
+import { BugFilingServiceProvider } from '../../../../../bug-filing/bug-filing-service-provider';
+import { BugFilingService } from '../../../../../bug-filing/types/bug-filing-service';
+import { IssueFilingButton, IssueFilingButtonDeps, IssueFilingButtonProps } from '../../../../../common/components/issue-filing-button';
+import { EnvironmentInfoProvider } from '../../../../../common/environment-info-provider';
+import { BugActionMessageCreator } from '../../../../../common/message-creators/bug-action-message-creator';
+import { NamedSFC } from '../../../../../common/react/named-sfc';
+import { UserConfigurationStoreData } from '../../../../../common/types/store-data/user-configuration-store';
+
+describe('IssueFilingButtonTest', () => {
+    let environmentInfoProviderMock: IMock<EnvironmentInfoProvider>;
+    let bugFilingServiceProviderMock: IMock<BugFilingServiceProvider>;
+    let bugActionMessageCreatorMock: IMock<BugActionMessageCreator>;
+    let userConfigurationStoreData: UserConfigurationStoreData;
+    const testKey: string = 'test';
+    let testBugService: BugFilingService;
+
+    beforeEach(() => {
+        testBugService = {
+            key: testKey,
+            displayName: 'TEST',
+            settingsForm: NamedSFC('testForm', props => <>Hello World</>),
+            isSettingsValid: () => true,
+            buildStoreData: testField => {
+                return { testField };
+            },
+            getSettingsFromStoreData: data => data[testKey],
+            createBugFilingUrl: () => 'test url',
+        };
+        userConfigurationStoreData = {
+            bugService: testKey,
+            bugServicePropertiesMap: {},
+        } as UserConfigurationStoreData;
+        environmentInfoProviderMock = Mock.ofType(EnvironmentInfoProvider);
+        bugFilingServiceProviderMock = Mock.ofType(BugFilingServiceProvider);
+        bugActionMessageCreatorMock = Mock.ofType(BugActionMessageCreator);
+        environmentInfoProviderMock
+            .setup(envp => envp.getEnvironmentInfo())
+            .returns(() => {
+                return {
+                    extensionVersion: '1',
+                    axeCoreVersion: '2',
+                    browserSpec: 'spec',
+                };
+            })
+            .verifiable();
```

Comment:
Usually, you don't need `verifiable` when the mock has been setup to return a value. This is specially through if that return value heavily influence the end result your asserting for. 
Are this values (versions and spec) being assert by another mock expectations? If that's true, you probably don't need the verifiable!

Reply:
So are we still 'verifying` this? why?

Reply:
I replied to this comment, but now it disappears!
So I think what you said make sense. In this particular case, we are not verifying the return value on other mocks, so I kept verifiable here.